rotkehlchen/chain/evm/accounting/structures.py

```diff
@@ -27,11 +28,39 @@ class TxAccountingTreatment(Enum):
     SWAP = 0
 
 
-class TxEventSettings(NamedTuple):
-    """Settings for an event generated by a decoder"""
-    taxable: bool
-    count_entire_amount_spend: bool
-    count_cost_basis_pnl: bool
-    method: Literal['acquisition', 'spend']
-    accounting_treatment: Optional[TxAccountingTreatment] = None
-    accountant_cb: Optional[AccountantCallback] = None
+class BaseEventSettins:
```

Comment:
As discussed this needs to be named properly.
```suggestion
class BaseEventSettings:
```